### PR TITLE
Fix build warning, ByteSize() is deprecated, use ByteSizeLong() instead

### PR DESCRIPTION
**Pull request by yangzhg**
Fix build warning, protobuf ByteSize() is deprecated, use ByteSizeLong() instead

### CODE REVIEW

**Comment by yangzhg** [comment]
build was failed due to `https://downloads.apache.org/thrift/0.11.0/thrift-0.11.0.tar.gz`  has moved to `https://archive.apache.org/dist/thrift/0.11.0/thrift-0.11.0.tar.gz`

**Comment by yangzhg** [comment]
build was failed due to `https://downloads.apache.org/thrift/0.11.0/thrift-0.11.0.tar.gz`  has moved to `https://archive.apache.org/dist/thrift/0.11.0/thrift-0.11.0.tar.gz`

File: src/brpc/policy/hulu_pbrpc_protocol.cpp [hunk 1]
```diff
@@ -152,7 +152,11 @@ inline void PackHuluHeader(char* hulu_header, int meta_size, int body_size) {
 template <typename Meta>
 static void SerializeHuluHeaderAndMeta(
     butil::IOBuf* out, const Meta& meta, int payload_size) {
+    #if GOOGLE_PROTOBUF_VERSION >= 3010000
```
[the remark this answers is not in the document]
Comment (yangzhg) [reply src/brpc/policy/hulu_pbrpc_protocol.cpp]: done

File: src/brpc/protocol.h [hunk 2]
```diff
@@ -56,6 +56,15 @@ class InputMessageBase;
 DECLARE_uint64(max_body_size);
 DECLARE_bool(log_error_text);
 
+// Get the serialized byte size of the protobuf message, 
+// different versions of protobuf have different methods
+
+#if GOOGLE_PROTOBUF_VERSION >= 3010000
+#define PROTOBUF_BYTE_SIZE(message) ((message).ByteSizeLong())
```
Comment (wwbmmm) [inline src/brpc/protocol.h:63]: 用宏容易产生命名冲突，可以改成inline函数吗

Comment (yangzhg) [reply src/brpc/protocol.h]: done

**Comment by serverglen** [comment]
LGTM

File: src/brpc/protocol.h [hunk 3]
```diff
@@ -56,6 +56,18 @@ class InputMessageBase;
 DECLARE_uint64(max_body_size);
 DECLARE_bool(log_error_text);
 
+// Get the serialized byte size of the protobuf message, 
+// different versions of protobuf have different methods
+// use template to avoid include `google/protobuf/message.h`
+template<typename T>
+inline uint32_t get_protobuf_byte_size(const T& message) {
```
Comment (wwbmmm) [inline src/brpc/protocol.h:63]: 全局函数驼峰命名，例如下面的ParsePbFromIOBuf

Comment (yangzhg) [reply src/brpc/protocol.h]: done

**Review by wwbmmm — APPROVED** [review]

File: src/brpc/protocol.h [hunk 4]
```diff
@@ -56,6 +56,18 @@ class InputMessageBase;
 DECLARE_uint64(max_body_size);
 DECLARE_bool(log_error_text);
 
+// Get the serialized byte size of the protobuf message, 
+// different versions of protobuf have different methods
+// use template to avoid include `google/protobuf/message.h`
+template<typename T>
+inline uint32_t GetProtobufByteSize(const T& message) {
+#if GOOGLE_PROTOBUF_VERSION >= 3010000
```
Comment (wasphin) [inline src/brpc/protocol.h:64]: 在不包含 protobuf 相关头文件的情况下, 直接用 `GOOGLE_PROTOBUF_VERSION` 有问题吧? 虽然不影响编译, 但结果恐怕不符合预期, 除非源文件中先包含 protobuf 相关头文件再包含此头文件.

Comment (wasphin) [reply src/brpc/protocol.h:64]: 目前看好像确实都是先包含的 protobuf 相关头文件.

Comment (wwbmmm) [reply src/brpc/protocol.h:64]: #include "brpc/options.pb.h"
这个应该包含了那个宏

Comment (wasphin) [reply src/brpc/protocol.h:64]: 看漏了…

**Pull request merged**